pandas/_typing.py

```diff
@@ -125,7 +125,17 @@
 JSONSerializable = Optional[Union[PythonScalar, List, Dict]]
 Frequency = Union[str, "DateOffset"]
 Axes = Collection[Any]
-RandomState = Union[int, ArrayLike, np.random.Generator, np.random.RandomState]
+
+if TYPE_CHECKING:
+    RandomState = Union[
+        int,
+        ArrayLike,
+        np.random.Generator,
+        np.random.BitGenerator,
+        np.random.RandomState,
+    ]
+else:
+    RandomState = Union[int, ArrayLike, np.random.Generator, np.random.RandomState]
```

Comment:
umm why is BitGenerator not in the 2nd part?

Reply:
BitGenerator isn't exposed until numpy >= 1.18. Have added a comment about this and removed the else clause (since that would mean we're not type checking anyway so the definition doesn't matter)